src/reducers/wallet.js

```diff
@@ -1,13 +1,28 @@
-import { WALLET_READ_SUCCESS, WALLET_CREATE_SUCCESS } from '../actions';
+import { WALLET_READ_REQUEST, WALLET_READ_SUCCESS, WALLET_READ_FAILURE, WALLET_CREATE_SUCCESS } from '../actions';
 
-const initialState = [];
+const initialState = { list: [] };
 
 const reducer = (state = initialState, { type, payload }) => {
   switch (type) {
+    case WALLET_READ_REQUEST:
+      return { ...state, isLoading: true };
     case WALLET_READ_SUCCESS:
-      return [...state, ...payload];
+      return {
+        ...state,
+        list: payload,
+        isLoading: false,
+      };
+    case WALLET_READ_FAILURE:
+      return {
+        ...state,
+        error: (payload.response || {}).error || payload.message,
+        isLoading: false,
+      };
     case WALLET_CREATE_SUCCESS:
-      return [...state, payload];
+      return {
+        ...state,
+        list: [...state.list, payload],
```

Comment:
Here we have another ES improvement, array destructuring. It will create a new array with all of members of `state.list` and append the `payload` (the newly created wallet).